File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        vector<int> values;

        // Traverse list1
        ListNode* curr = list1;
        while (curr != nullptr) {
            values.push_back(curr->val);
            curr = curr->next;
        }

        // Traverse list2
        curr = list2;
        while (curr != nullptr) {
            values.push_back(curr->val);
            curr = curr->next;
        }

        // Sort all values
        sort(values.begin(), values.end());

        // Create new sorted list from vector
        ListNode* dummy = new ListNode(-1);
        curr = dummy;
        for (int val : values) {
            curr->next = new ListNode(val);
            curr = curr->next;
        }

        return dummy->next;
    }
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
 //m1 (n+m)*log(n+m)
//  Sc: O(n+m)
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ArrayList<Integer>  al = new ArrayList<>();
        ListNode curr = list1;
        while(curr!=null){
            al.add(curr.val);
            curr=curr.next;
        }
        curr = list2;
        while(curr!=null){
            al.add(curr.val);
            curr=curr.next;
        }
        Collections.sort(al);
        ListNode dummy= new ListNode(-1);
        curr=dummy;
       for(int i= 0; i<al.size(); i++){
            curr.next = new ListNode(al.get(i));
            curr=curr.next;
       }
       return dummy.next;
    }
}